src/hooks/useBorrowRepayModal/Modal/Borrow/index.tsx

```diff
@@ -87,67 +87,81 @@ export const BorrowForm: React.FC<BorrowFormProps> = ({
 
   return (
     <AmountForm onSubmit={onSubmit} maxAmount={limitTokens}>
-      {({ values, dirty, isValid, errors }) => (
-        <>
-          {pool.isIsolated && (
-            <IsolatedAssetWarning
+      {({ values, dirty, isValid, errors }) => {
+        const isHighRiskBorrow =
+          new BigNumber(values.amount).isGreaterThanOrEqualTo(safeLimitTokens) &&
+          new BigNumber(values.amount).isLessThanOrEqualTo(limitTokens);
+
+        const enableLabel = isHighRiskBorrow
+          ? t('borrowRepayModal.borrow.submitButtonHighRisk')
+          : t('borrowRepayModal.borrow.submitButton');
```

Comment:
I'm generally not a fan of declaring variables inside the return statement of a component as I feel it obfuscates the logic. An alternative could be to use a memoized callback:
```tsx
const isHighRiskBorrow = useCallback(
  (amount: string) =>
    new BigNumber(amount).isGreaterThanOrEqualTo(safeLimitTokens) &&
    new BigNumber(amount).isLessThanOrEqualTo(limitTokens),
  [limitTokens, safeLimitTokens],
);
```

Then you can use it inside the return statement.
The two advantages I see are that we keep the logic out of the return statement and that these constants don't get redeclared with every render. Let me know your thoughts though in case you prefer the approach you took originally!

Reply:
I created a new component and moved all the related logic to it, this way we avoid having this variable inside the return statement and `BorrowForm` is cleaner now, let me know if you'd like additional changes 👍🏻